WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/ProductRowViewModel.swift

```diff
@@ -0,0 +1,96 @@
+import Foundation
+import Yosemite
+
+/// View model for `ProductRow`.
+///
+final class ProductRowViewModel: ObservableObject, Identifiable {
+    /// Product ID
+    /// Required by SwiftUI as a unique identifier
+    ///
+    let id: Int64
+
+    private let currencyFormatter: CurrencyFormatter
+
+    /// Whether the product quantity can be changed.
+    /// Controls whether the stepper is rendered.
+    ///
+    let canChangeQuantity: Bool
+
+    /// Product to display
+    ///
+    let product: Product
+
+    /// Label showing product name
+    ///
+    var nameLabel: String {
+        product.name
+    }
+
+    /// Label showing product stock status and price.
+    ///
+    var stockAndPriceLabel: String {
+        let stockLabel = createStockText()
+        let priceLabel = createPriceText()
+
+        return [stockLabel, priceLabel]
+            .compactMap({ $0 })
+            .joined(separator: " • ")
+    }
+
+    /// Label showing product SKU
+    ///
+    var skuLabel: String {
+        guard let sku = product.sku, sku.isNotEmpty else {
+            return ""
+        }
+        return String.localizedStringWithFormat(Localization.skuFormat, sku)
+    }
+
+    /// Quantity of product in the order
+    ///
+    var quantity: Int64 = 1
```

Comment:
Are you planing to mark this as a @Published variable later? maybe we edit quantity is implemented?

Reply:
I was planning to think about that when I implemented quantity editing, but now that you've pointed it out I'll make the change now so I don't forget later and get confused. 😅